scripts/ci-deb-build-test

```diff
@@ -84,6 +84,7 @@ su testuser -c '
     fi
   fi
   go version
+  export GOTOOLCHAIN=auto
```

Comment:
I am skeptical about this.  The script already has its own code for downloading & compiling a newer go version when needed.  What happens when it is not set (here and for rpm)?

Reply:
@DrDaveD 
see the error in the pipeline
https://github.com/apptainer/apptainer/actions/runs/11798829974/job/32865935843?pr=2614
it'll use container's golang toolchai.

Reply:
If you look earlier in that output, it is already downloading the source and binary of go 1.22.5.   That is required to build the resulting source rpm on EPEL/Fedora (the output of that ci script is also used for the released packages on github), because online downloads are not allowed on that build system.   There's something else wrong.  We must not set GOTOOLCHAIN=auto.

I reproduced the problem interactively under docker containers.  The problem is that the first mconfig stores the go path in the Makefile.  The best solution I can come up with is to re-run mconfig after the go binary is downloaded:
```
diff --git a/scripts/ci-rpm-build-test b/scripts/ci-rpm-build-test
index 90c5b421d..59da62771 100755
--- a/scripts/ci-rpm-build-test
+++ b/scripts/ci-rpm-build-test
@@ -54,6 +54,8 @@ su testuser -c '
       curl -f -L -sS https://golang.org/dl/$GOBIN | tar -xzf - -C /local
       PATH=/local/go/bin:$PATH
     fi
+    # re-do the mconfig to locate the new go
+    ./mconfig --only-rpm
   fi
   go version
   ./scripts/download-dependencies
```

Debian does not have the same problem since it doesn't need to use mconfig to build a .spec file.  I removed `export GOTOOLCHAIN=auto` from scripts/ci-deb-build-test and it worked fine in my test.  Did you see a failure there?

Reply:
thanks! looks like it works.